src/libraries/Microsoft.Extensions.Logging.Console/src/JsonConsoleLogFormatter.cs

```diff
@@ -0,0 +1,200 @@
+{
+    internal class JsonConsoleLogFormatter : ConsoleFormatter, IDisposable
+    {
+        private IDisposable _optionsReloadToken;
+
+        [ThreadStatic]
+        private static StringBuilder _logBuilder;
+
+        public JsonConsoleLogFormatter(IOptionsMonitor<JsonConsoleFormatterOptions> options)
+            : base (ConsoleFormatterNames.Json)
+        {
+            FormatterOptions = options.CurrentValue;
+            ReloadLoggerOptions(options.CurrentValue);
+            _optionsReloadToken = options.OnChange(ReloadLoggerOptions);
+        }
+
+        private string WriteJson(LogLevel logLevel, string category, int eventId, string message, Exception exception, IExternalScopeProvider scopeProvider)
+        {
+            const int DefaultBufferSize = 1024;
+            var output = new ArrayBufferWriter<byte>(DefaultBufferSize);
+            using (var writer = new Utf8JsonWriter(output, FormatterOptions.JsonWriterOptions))
+            {
+                writer.WriteStartObject();
+
+
+                string timestamp = null;
+                var timestampFormat = FormatterOptions.TimestampFormat;
+                if (timestampFormat != null)
+                {
+                    var dateTime = FormatterOptions.UseUtcTimestamp ? DateTime.UtcNow : DateTime.Now;
+                    timestamp = dateTime.ToString(timestampFormat);
+                }
+                writer.WriteString("timestamp", timestamp);
+                writer.WriteNumber("eventId", eventId);
+                writer.WriteString("logLevel", GetLogLevelString(logLevel));
+                writer.WriteString("category", category);
+                writer.WriteString("message", message);
+
+                if (exception != null)
+                {
+                    writer.WriteStartObject("exception");
+                    writer.WriteString("message", exception.Message.ToString());
+                    writer.WriteString("type", exception.GetType().ToString());
+                    writer.WriteStartArray("stackTrace");
+                    if (exception?.StackTrace != null)
+                    {
+                        foreach (var xx in exception?.StackTrace?.Split(Environment.NewLine))
+                        {
+                            JsonSerializer.Serialize<string>(writer, xx);
+                        }
+                    }
+                    writer.WriteEndArray();
+                    writer.WriteNumber("hResult", exception.HResult);
+                    writer.WriteEndObject();
+                }
+
+                GetScopeInformation(writer, scopeProvider);
+
+                writer.WriteEndObject();
+
+                writer.Flush();
+            }
+            return Encoding.UTF8.GetString(output.WrittenMemory.Span);
+        }
+
+        public override void Write<TState>(in LogEntry<TState> logEntry, IExternalScopeProvider scopeProvider, TextWriter textWriter)
+        {
+            if (textWriter is StringWriter stringWriter)
```

Comment:
This check shouldn't be there. You should only do a type check like this for an optimization, it should still work if the writer isn't a string writer.

Reply:
TextWriter needs to have set color APIs that is why I am casting. This condition will always be true because the only called is in ConsoleLogger